File: RemoveDuplicatesFromSortedList/RemoveDuplicatesFromSortedList.cpp
/**
* Definition for singly-linked list.
*/
struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(nullptr) {}
};

class Solution {
public:
	ListNode* deleteDuplicates(ListNode* head) {
		if (head == nullptr || head->next == nullptr)
			return head;
		ListNode* cur = head;
		ListNode* tmp = cur->next;
		while (cur != nullptr) {
			if (tmp == nullptr) {
				cur->next = tmp;
				cur = cur->next;
			}
			else {
				if (cur->val == tmp->val)
					tmp = tmp->next;
				else
				{
					cur->next = tmp;
					cur = tmp;
				}
			}
			
		}
		return head;
	}
};

File: RemoveDuplicatesFromSortedList/I/RemoveDuplicatesFromSortedList.py
# Definition for singly-linked list.
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head == None:
        	return head
        cur = head
        tmp = cur.next
        while cur != None:
        	if tmp == None:
        		cur.next = tmp
        		cur = cur.next
        	else:
        		if cur.val == tmp.val:
        			tmp = tmp.next
        		else:
        			cur.next = tmp
        			cur = cur.next
        return head